diesel_derives/src/for_each_tuple.rs

```diff
@@ -0,0 +1,977 @@
+use std::ops::Range;
+
+use syn::parse::Parse;
+use syn::spanned::Spanned;
+use syn::{Attribute, Ident, Type, TypeParam};
+
+#[cfg(feature = "128-column-tables")]
+pub const DEFAULT_SIZE: i32 = 128;
+
+#[cfg(all(feature = "64-column-tables", not(feature = "128-column-tables")))]
+pub const DEFAULT_SIZE: i32 = 64;
+
+#[cfg(all(
+    feature = "32-column-tables",
+    not(feature = "128-column-tables"),
+    not(feature = "64-column-tables")
+))]
+pub const DEFAULT_SIZE: i32 = 32;
+
+#[cfg(all(
+    not(feature = "64-column-tables"),
+    not(feature = "128-column-tables"),
+    not(feature = "32-column-tables")
+))]
+pub const DEFAULT_SIZE: i32 = 16;
+
+macro_rules! timed_block {
+    ($start: expr, $text: expr, $debug: expr) => {
+        if $debug {
+            let elapsed = $start.elapsed();
+            eprintln!(
+                "    {}: {}.{:06} s",
+                $text,
+                elapsed.as_secs(),
+                elapsed.subsec_micros()
+            );
+        }
+    };
+}
+
+macro_rules! start_timed_block {
+    ($name: expr, $debug: expr) => {{
+        if $debug {
+            eprintln!("    {}", $name);
+        }
+        std::time::Instant::now()
+    }};
+}
```

Comment:
I guess these are just temporary?

Reply:
Yes they are temporary. I've left them there for now as I'm not that happy with the performance of this proc macro attribute. I would have guessed that this approach can be much faster for large tuple sizes than the old macro by example based implementation, but that's unfortunately not the case. This macros are there to have something to measure what takes that much time.